ganttproject-tester/test/net/sourceforge/ganttproject/test/task/TestTaskCompletionPercentage.java

```diff
@@ -124,6 +124,54 @@ public void testCompletionWithMilestones() {
     //
     RecalculateTaskCompletionPercentageAlgorithm alg = taskManager.getAlgorithmCollection().getRecalculateTaskCompletionPercentageAlgorithm();
     alg.run(supertask);
-    assertEquals("Unexpected completion percentage of supertask=" + supertask, 75, supertask.getCompletionPercentage());
+    assertEquals("Unexpected completion percentage of supertask=" + supertask, 50, supertask.getCompletionPercentage());
+  }
+
+  public void testCompletionWithNestedTasksDepthGreaterThan1AndGapsInCalender() {
+    TaskManager taskManager = getTaskManager();
+    Task project = taskManager.createTask();
+    project.setProjectTask(true);
+    Task supertask = taskManager.createTask();
+
+    Task supertask_0 = taskManager.createTask();
+    Task supertask_0_0 = taskManager.createTask();
+    Task supertask_0_1 = taskManager.createTask();
+    Task supertask_1 = taskManager.createTask();
+    Task supertask_1_0 = taskManager.createTask();
+    Task supertask_1_1 = taskManager.createTask();
+
+    supertask_0_0.setStart(CalendarFactory.createGanttCalendar(2000, 01, 10));
```

Comment:
I recommend using `TaskBuilder` interface, like this:
```
Task task_0_0 = taskManager.newTaskBuilder()
    .withStartDate(TestSetupHelper.newMonday().getTime())
    .withDuration(taskManager.createLength(1))
    .withCompletion(100)
    .withParent(supertask_0)
    .build();
```

This will make the test code more readable. 

Please pay attention that in the code end date is start date + 1.

Reply:
Nice. I just used the available code within the existing tests.